### PR TITLE
app/peerinfo: added builder_api_enabled metric

### PR DESCRIPTION
**Pull request by pinebit**
Adds `app_peerinfo_builder_api_enabled` metric (gauge) indicating if the `peer` has builder API enabled.

category: feature
ticket: #2879

### CODE REVIEW

**Review by dB2510 — CHANGES_REQUESTED** [review]
you will need to add mev_enabled field to peerinfo protobuf message in `app/peerinfo/peerinfopb/v1/peerinfo.proto` to send/receive this info cluster peers

**Comment by pinebit** [comment]
> you will need to add mev_enabled field to peerinfo protobuf message in `app/peerinfo/peerinfopb/v1/peerinfo.proto` to send/receive this info cluster peers

Thank you, done.

File: app/peerinfo/metrics.go [hunk 1]
```diff
@@ -53,4 +53,11 @@ var (
 		Name:      "version_support",
 		Help:      "Set to 1 if the peer's version is supported by (compatible with) the current version, else 0 if unsupported.",
 	}, []string{"peer"})
+
+	peerMevEnabledGauge = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "app",
+		Subsystem: "peerinfo",
+		Name:      "mev_enabled",
+		Help:      "Set to 1 if mev is enabled on this peer, else 0 if disabled.",
```
Comment (OisinKyne) [inline app/peerinfo/metrics.go:61]: Although this is about MEV, can we call this `builder_api_enabled` to be in line with the env var/flag naming? 

```suggestion
		Name:      "builder_api_enabled",
		Help:      "Set to 1 if builder-api is enabled on this peer, else 0 if disabled.",
```

Won't comment through the rest of this PR for simplicities sake.

Comment (pinebit) [reply app/peerinfo/metrics.go:61]: Accepted. Thank you!

**Comment by pinebit** [comment]
Added Builder API column as following:
image

**Review by dB2510 — APPROVED** [review]
LGTM 🚀

**Review by gsora — APPROVED** [review]
10/10, one of the most useful information for debugging purposes is now easily accessible!

**Pull request merged**